### PR TITLE
Add seldon deployment predictor parameters and resource requirements

### PR DESCRIPTION
**Pull request by d-lowl**
## Describe changes
I implemented passing optional list of parameters to predictors in Seldon Deployments (within Seldon integration).

Add parameters to `SeldonDeploymentPredictiveUnit` and `SeldonDeploymentConfig`, and pass these parameters all the way to the seldon deployment manifest.
Add resource requirements to `SeldonDeploymentPredictors` and `SeldonDeploymentConfig`, and pass these parameters all the way to the seldon deployment manifest.
Document usage.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes. _not applicable, no test suite_

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by d-lowl** [comment]
A feature request has been made earlier today https://zenml.hellonext.co/p/in-seldon-integration-allow-for-predictor-parameters-to-be-passed

File: src/zenml/integrations/seldon/seldon_client.py [hunk 1]
```diff
@@ -93,6 +127,9 @@ class SeldonDeploymentPredictiveUnit(BaseModel):
     ] = SeldonDeploymentPredictiveUnitType.MODEL
     implementation: Optional[str]
     modelUri: Optional[str]
+    parameters: List[SeldonDeploymentPredictorParameter] = Field(
+        default_factory=list
+    )
```
Comment (safoinme) [inline src/zenml/integrations/seldon/seldon_client.py:132]: Is the `parametrs` required for every deployment? or can we make this:
```suggestion
    parameters: Optional[List[SeldonDeploymentPredictorParameter]]
```

File: src/zenml/integrations/seldon/seldon_client.py [hunk 2]
```diff
@@ -282,6 +322,10 @@ def build(
         model_uri: Optional[str] = None,
         model_name: Optional[str] = None,
         implementation: Optional[str] = None,
+        parameters: List[SeldonDeploymentPredictorParameter] = Field(
+            default_factory=list
+        ),
```
Comment (safoinme) [inline src/zenml/integrations/seldon/seldon_client.py:327]: same as above? can this be `Optional`

**Review by safoinme — CHANGES_REQUESTED** [review]
Awesome Work! I just wanted to take a moment to thank you for your contribution. The changes you made will greatly improve the seldon deployment story in ZenML 

I left a comment regarding the `parameters` attribute.

**Comment by safoinme** [comment]
@d-lowl can you please add the word `engineResources` to the `.pyspelling-ignore-words` file to fix check spelling error.

File: src/zenml/integrations/seldon/seldon_client.py (hunk 1 above)
Comment (d-lowl) [reply src/zenml/integrations/seldon/seldon_client.py:132]: It is not required indeed, and can be optional. It cannot be null in the k8s manifest though, so I need to check how optional parameters are serialised.

Comment (d-lowl) [reply src/zenml/integrations/seldon/seldon_client.py]: These were made optional now. As far as I can see from manual testing, it still behaves as expected

**Comment by d-lowl** [comment]
Thanks for the review comments, they've been addressed now. Good to merge when you are ready

**Review by fa9r — APPROVED** [review]

**Pull request merged**